lib/rules/valid-api.js

```diff
@@ -68,17 +68,15 @@ function validatePropertyName(property, context) {
         });
     }
 
-    if (name.startsWith('data')) {
+    if (RESERVED_PUBLIC_PROPERTIES.has(name)) {
         context.report({
             node: property,
-            message: `Invalid public property "${name}". Properties starting with "data" are reserved properties.`,
+            message: `Invalid public property "${name}". This property name is a reserved property.`,
         });
-    }
-
-    if (RESERVED_PUBLIC_PROPERTIES.has(name)) {
+    } else if (name.startsWith('data') && name !== 'data') {
```

Comment:
```suggestion
    } else if (/^data[A-Z]/.test(name)) {
```

Reply:
Can we make the above change and also make it obvious (either via comment or test) that the purpose of this rule is to avoid property names that can't be set through content attributes?